File: common/src/main/java/dev/obscuria/fragmentum/core/CoreFragmentumClient.java
package dev.obscuria.fragmentum.core;

import dev.obscuria.fragmentum.api.v1.client.FragmentumClientRegistry;
import dev.obscuria.fragmentum.core.v1.client.AllOfClientTooltipComponent;
import dev.obscuria.fragmentum.core.v1.common.AllOfTooltipComponent;
import org.jetbrains.annotations.ApiStatus;

@ApiStatus.Internal
public final class CoreFragmentumClient
{
    public static void init()
    {
        FragmentumClientRegistry.registerTooltipComponent(AllOfTooltipComponent.class, AllOfClientTooltipComponent::new);
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/core/CoreFragmentumClient.kt
package dev.obscuria.fragmentum.core

import dev.obscuria.fragmentum.api.client.FragmentumClientRegistry
import dev.obscuria.fragmentum.core.client.AllOfClientTooltipComponent
import dev.obscuria.fragmentum.core.world.AllOfTooltipComponent
import org.jetbrains.annotations.ApiStatus.Internal

@Internal
object CoreFragmentumClient {

    fun init() {
        FragmentumClientRegistry.registerTooltipComponent(AllOfTooltipComponent::class.java) {
            AllOfClientTooltipComponent(it)
        }
    }
}